libs/SalesforceSDKCore/SalesforceSDKCore/Classes/SmartStore/SFSmartStoreInspectorViewController.m

```diff
@@ -88,14 +88,28 @@ + (SFSmartStoreInspectorViewController *) sharedInstance
 
 #pragma mark - Present / dimiss
 
-+ (void) present
++ (void) present:(UIViewController*)currentViewController
 {
-    [[SFRootViewManager sharedManager] pushViewController:[SFSmartStoreInspectorViewController sharedInstance]];
+    if (![NSThread isMainThread]) {
+        dispatch_async(dispatch_get_main_queue(), ^{
+            [SFSmartStoreInspectorViewController present:currentViewController];
+        });
+        return;
+    }
+
+    [currentViewController presentViewController:[SFSmartStoreInspectorViewController sharedInstance] animated:NO completion:nil];
```

Comment:
I see your main thread, and raise you:

        dispatch_async(dispatch_get_main_queue(), ^{ [self.viewController presentViewController:myViewController animated:YES completion:NULL });

:)  I'm okay keeping the pattern for the reasons you stated.  My only point is that iOS developers learn this pattern from day one, and use it every day.  I don't feel like we should be getting into the practice of refactoring or otherwise changing development patterns that are already commonplace in the development environments we support.